### PR TITLE
Implement triggers for vuln det scans

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
|#7924|

## Description
This PR implements the triggers to decide if Vulnerability Detector must generate a BASELINE_SCAN, a PARTIAL_SCAN or a FULL_SCAN.
The BASELINE_SCAN is performed when vuln_det is enabled for the first time and LAST_FULL_SCAN is 0.
The FULL_SCAN is performed when there is a change in the feed related to the scanned agent and the minimum time for full_scans is reached.
The PARTIAL_SCAN is done as the default action.

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)

### CODE REVIEW

**Comment by palaciosjeremias** [comment]
Scan-build test:
image

**Comment by palaciosjeremias** [comment]
Valgrind on UT:
image

File: src/wazuh_db/schema_agents.sql [hunk 1]
```diff
@@ -355,11 +355,11 @@ CREATE TABLE IF NOT EXISTS sca_check_compliance (
...
 
 CREATE TABLE IF NOT EXISTS vuln_metadata (
-    LAST_SCAN INTEGER,
-    WAZUH_VERSION TEXT,
+    LAST_PARTIAL_SCAN INTEGER,
+    LAST_FULL_SCAN INTEGER,
     HOTFIX_SCAN_ID TEXT
 );
-INSERT INTO vuln_metadata (LAST_SCAN, WAZUH_VERSION, HOTFIX_SCAN_ID)
+INSERT INTO vuln_metadata (LAST_PARTIAL_SCAN, LAST_FULL_SCAN, HOTFIX_SCAN_ID)
```
Comment (pereyra-m) [inline src/wazuh_db/schema_agents.sql:362]: Same comment about the column types.

**Review by pereyra-m — APPROVED** [review]
LGTM!

File: src/wazuh_db/schema_upgrade_v8.sql [hunk 2]
```diff
@@ -28,7 +28,17 @@ CREATE INDEX IF NOT EXISTS cves_id ON vuln_cves (cve);
...
-UPDATE vuln_metadata SET LAST_SCAN = '0';
+DROP TABLE IF EXISTS vuln_metadata;
+
+CREATE TABLE IF NOT EXISTS vuln_metadata (
+    LAST_PARTIAL_SCAN INTEGER,
+    LAST_FULL_SCAN INTEGER,
+    HOTFIX_SCAN_ID TEXT
+);
+INSERT INTO vuln_metadata (LAST_PARTIAL_SCAN, LAST_FULL_SCAN, HOTFIX_SCAN_ID)
+    SELECT '0', '0', '0' WHERE NOT EXISTS (
```
[the remark this answers is not in the document]
Comment (palaciosjeremias) [reply src/wazuh_db/schema_upgrade_v8.sql:39]: You are right. This is legacy and looks like it works anyway. But these values should be **0** as an integer.
Fixed

File: src/wazuh_db/schema_agents.sql (hunk 1 above)
Comment (palaciosjeremias) [reply src/wazuh_db/schema_agents.sql:362]: Fixed

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 3]
```diff
@@ -290,6 +288,57 @@ STATIC void wm_vuldet_json_msu_parser_vuln(cJSON *json_feed, vu_msu_vul_entry **
...
+/**
+ * @brief Checks if the feed(s) for an agent changed since last scan.
+ *
+ * @param agent The scaned agent data structure.
+ * @param last_scan The time of the last scan.
+ *
+ * @return TRUE if any of the feeds of the agent has been modified after the last scan.
+ *         FALSE if all the feeds of the agent hasn't been modified since the last scan.
+ */
+STATIC bool wm_vuldet_feed_changed (update_node** feeds, scan_agent* agent, time_t last_scan);
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:340]:
```suggestion
STATIC bool wm_vuldet_feed_changed(update_node** feeds, scan_agent* agent, time_t last_scan);
```

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Please, apply changes according to the comments.

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**